File: hw/palacios/u3/HW18Singleton/HW18Singleton-JS/model/USTax.js
class USTax {
    constructor(taxPercentage) {
        this.taxPercentage = taxPercentage;
    }

    static getInstance() {
        if (!USTax.instance) {
            USTax.instance = new USTax(0.2);
        }
        return USTax.instance;
    }

    static setInstance(newInstance) {
        if (newInstance) {
            USTax.instance = newInstance;
        } else {
            console.log("Cannot set null instance");
        }
    }

    getTaxPercentage() {
        return this.taxPercentage;
    }

    setTaxPercentage(taxPercentage) {
        this.taxPercentage = taxPercentage;
    }
}

module.exports = USTax;

File: hw/maisincho/u3/HW18Singleton/Singleton/src/main/java/ec/edu/espe/singleton/model/USTax.java

package ec.edu.espe.singleton.model;

/**
 *
 * @author Isaac Maisincho Crafters_Market DCCO ESPE
 */
public class USTax {

    private static USTax instance;
    private float taxPercentage;

    public USTax(float taxPercentage) {
        this.taxPercentage = taxPercentage;
    }

    public static USTax getInstance() {
        if (instance == null) {
            instance = new USTax(0.2F);
        }
        return instance;

    }

    public static void setInstance(USTax newInstance) {
        if (newInstance != null) {
            instance = newInstance;
        } else {
            System.out.println("Cannot set null instance ");
        }
    }

    public float getTaxPercentage() {
        return taxPercentage;
    }

    public void setTaxPercentage(float taxPercentage) {
        this.taxPercentage = taxPercentage;
    }

}
